sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -161,8 +161,14 @@ def build_task_spec_for_task(
                         input_name].component_input_artifact = (
                             component_input_artifact)
             else:
-                raise RuntimeError(
-                    f'Artifacts must be produced by a task. Got {input_value}.')
+                component_input_artifact = input_value.full_name
```

Comment:
I think we can fairly cleanly merge the `isinstance(input_value, pipeline_channel.PipelineArtifactChannel):` and `isinstance(input_value, pipeline_channel.PipelineParameterChannel):` branches here with only a few uses of `getattr` to access the correct attribute. WDYT?

Reply:
SGTM -- just wanted to check that it was considered :)